doc/development/tutorials/helloworld.rst

```diff
@@ -56,30 +72,33 @@ Open :file:`helloworld.py` and paste the following code in it:
        app.add_directive("helloworld", HelloWorld)
 
 
-Some essential things are happening in this example, and you will see them
-in all directives:
+Some essential things are happening in this example, and you will see them for
+all directives.
 
-.. rubric:: Directive declaration
+.. rubric:: The directive class
 
-Our new directive is declared in the ``HelloWorld`` class, it extends
-docutils_' ``Directive`` class. All extensions that create directives
-should extend this class.
+Our new directive is declared in the ``HelloWorld`` class.
 
-.. rubric:: ``run`` method
+.. code-block:: python
 
-This method is a requirement and it is part of every directive. It contains
-the main logic of the directive and it returns a list of docutils nodes to
-be processed by Sphinx.
+   class HelloWorld(Directive):
 
-.. seealso::
+       def run(self):
+           paragraph_node = nodes.paragraph(text='Hello World!')
+           return [paragraph_node]
 
-   :doc:`todo`
+This class extends the docutils_' ``Directive`` class. All extensions that
+create directives should extend this class.
```

Comment:
I don't understand the difference, to be honest. In fact, I thought we had removed that in favour of the docutils `Directive` class, but maybe I'm thinking of something else. Do you see any reason to do this here or in a later tutorial?

Reply:
Hmm? removed? Now we're using `SphinxDirective` class to make custom directives.
https://github.com/sphinx-doc/sphinx/blob/6bdac80b5fb75641f7cd6f4b8c47bf6241360e2c/sphinx/directives/__init__.py#L49

I believe `SphinxDirective` is useful to develop Sphinx extension. So it would be better to introduce it in this tutorial.

Reply:
As noted below, I'm still not sure why to use this. Could you elaborate on why it's useful. I can add a note saying we're extending from the docutils directive class but one should extend from the Sphinx one if they need access to `self.config` or the likes?

Reply:
With docutils' Directive class, you need to write a complex code to access Sphinx objects (ex. `app`, `env`, `config`, `builder`).
For example, in case of a directive would like to access `myext_config` confval, you need to write following code:
```
class MyDirective(docutils.parsers.rst.Directive):
    def run(self):
        if self.state.document.settings.env.config.myext_config:
            ...
```

It is hard to write and understand for all developers.

With `SphinxDirective` class, you can access it simply with `self.config`:
```
class MyDirective(sphinx.util.docutils.SphinxDirective):
    def run(self):
        if self.config.myext_config:
            ...
```

If custom directives does not access to Sphinx objects, our `SphinxDirective` is not needed. So helloworld and todo extensions would not need to inherit it. But developers would need to know the way to access to Sphinx objects in real case. So I think here is good point to let them know about `SphinxDirective`.

Reply:
I've added this to the todo case. Let me know if you think this is still unclear.